devito/passes/iet/misc.py

```diff
@@ -146,5 +147,12 @@ def is_on_device(maybe_symbol, gpu_fit, only_writes=False):
             expressions = FindNodes(Expression).visit(iet)
             functions &= {i.write for i in expressions}
 
+    if 'all-fallback' in gpu_fit and any(f.is_TimeFunction and f.save is not None
+                                         for f in functions):
+        warning("gpu-fit option didn't include save-full TimeFunctions [...] which "
```

Comment:
I would:

```
fsave = [f for f in functions if f.is_TimeFunction and f.save is not None]
```

then

```
if 'all-fallback in gpu_fit and  fsave:
    warning(".... spell out the fsave functions here....")
```

also, maybe we can drop

```
make sure their runtime size honors this
```

(because the other obvious alternative would be to start using `gpu-fit`